lib/orbf/rules_engine/fetch_data/fetch_data.rb

```diff
@@ -6,57 +6,23 @@ def initialize(dhis2_connection, package_arguments)
         @dhis2_connection = dhis2_connection
       end
 
+      # TODO SMS ask piet ;)
+      #    - what to do it data is coming from data values sets to
+      #       - drop the one from analytics, need to find "duplicates" ?
```

Comment:
I think data value sets > analytics, so when this happens we should default to pick the data value set value instead of the analytic value. That said, I would also be okay, to 'let it play out', maybe raise a warning but let the last value that comes in "win"

Reply:
this has been solved by the "filtering" code, respecting the activity_state definition in ActivityVariablesBuilder

Reply:
no guess, no magic and might be used to compare the analytics and the dataValueSets values

Reply:
Even better